**<center><b><u>Project 2: - Tidy, Untidy Data</u></center></b>**
===========================================

<b><u>Tasks:</b></u><br>
<br>
Choose any three of the “wide” datasets identified in the Week 5 Discussion items. (You may use your own dataset; please don’t use my Sample Post dataset, since that was used in your Project 1 assignment!) For each of the three chosen datasets:
<ol>
<li>Create a .CSV file (or optionally, a MySQL database!) that includes all of the information included in the dataset. You’re encouraged to use a “wide” structure similar to how the information appears in the discussion item, so that you can practice tidying and transformations as described below.
<br>
<br>
<li>Read the information from your .CSV file into Python, and use pandas as needed to tidy and transform your data. [Most of your grade will be based on this step!]
<br>
<br>
<li>Perform the analysis requested in the discussion item.
<br>
<br>
<li>Your code should be in an Jupyter Notebook, posted to your GitHub repository, and
should include narrative descriptions of your data cleanup work, analysis, and
conclusions.
</ol>  

**<b><u>Dataset 1 - Religious Data Analysis:</u></b>**
===========================================

**<center><b>To analyze the data in qustion we must import the pandas and numpy libraries. The following code will import both libaries.<center><b>**

In [1]:
import pandas as pd
import numpy as np

**<center><b>Next, the code below will import the csv files using pandas, drop the column named Unnamed:0,<br> convert the data into a datframe, and generate the output.<center><b>**

In [2]:
covid = pd.read_csv('relinc.csv', index_col='religion')
#print(covid.columns)
covid.drop('Unnamed: 0', axis=1, inplace=True)
covid = pd.DataFrame(covid)
covid

<10k  10-20k  20-30k  30-40k  40-50k  50-75k  75-100k  100-150k  \
religion                                                                    
Agnostic    27      34      60      81      76     137      122       109   
Atheist     12      27      37      52      35      70       73        59   
Buddhist    27      21      30      34      33      58       62        39   
Catholic  4418     617     732     670     638    1116      949       792   
refused     15      14      15      11      10      35       21        17   

          >150k  refused  
religion                  
Agnostic     84       96  
Atheist      74       76  
Buddhist     53       54  
Catholic    633     1489  
refused      18      116

**<center><b>Next, we will rename the first and last column titles so we can understand the variables more clearly.<center><b>**

In [3]:
covid.rename(columns = {'<10k':'Income: <10k','refused':'Unknown - Refused to Answer'}, inplace = True)                               
covid

Income: <10k  10-20k  20-30k  30-40k  40-50k  50-75k  75-100k  \
religion                                                                  
Agnostic            27      34      60      81      76     137      122   
Atheist             12      27      37      52      35      70       73   
Buddhist            27      21      30      34      33      58       62   
Catholic          4418     617     732     670     638    1116      949   
refused             15      14      15      11      10      35       21   

          100-150k  >150k  Unknown - Refused to Answer  
religion                                                
Agnostic       109     84                           96  
Atheist         59     74                           76  
Buddhist        39     53                           54  
Catholic       792    633                         1489  
refused         17     18                          116

**<center><b>Let's add a column which depicts the total number of individuals in each denomination category.<center><b>**

In [4]:
covid['Total'] = covid.sum(axis=1)
covid

Income: <10k  10-20k  20-30k  30-40k  40-50k  50-75k  75-100k  \
religion                                                                  
Agnostic            27      34      60      81      76     137      122   
Atheist             12      27      37      52      35      70       73   
Buddhist            27      21      30      34      33      58       62   
Catholic          4418     617     732     670     638    1116      949   
refused             15      14      15      11      10      35       21   

          100-150k  >150k  Unknown - Refused to Answer  Total  
religion                                                       
Agnostic       109     84                           96    826  
Atheist         59     74                           76    515  
Buddhist        39     53                           54    411  
Catholic       792    633                         1489  12054  
refused         17     18                          116    272

**<center><b>Let's do some analaysis on the data. First let's find the total number of individuals in the dataset:<center><b>**

In [5]:
total_sample_count = covid['Total'].sum()
total_sample_count

14078

**<center><b>Next, we will create a column which depicts the percent of individuals per denomination:<center><b>**

In [6]:
covid.insert(11,'Religious Population Percent', (covid.Total / total_sample_count) * 100)
pd.options.display.float_format = '{:.1f}%'.format
covid

Income: <10k  10-20k  20-30k  30-40k  40-50k  50-75k  75-100k  \
religion                                                                  
Agnostic            27      34      60      81      76     137      122   
Atheist             12      27      37      52      35      70       73   
Buddhist            27      21      30      34      33      58       62   
Catholic          4418     617     732     670     638    1116      949   
refused             15      14      15      11      10      35       21   

          100-150k  >150k  Unknown - Refused to Answer  Total  \
religion                                                        
Agnostic       109     84                           96    826   
Atheist         59     74                           76    515   
Buddhist        39     53                           54    411   
Catholic       792    633                         1489  12054   
refused         17     18                          116    272   

          Religious Population Percent  
religion                                
Agnostic                          5.9%  
Atheist                           3.7%  
Buddhist                          2.9%  
Catholic                         85.6%  
refused                           1.9%

**<center><b>Based on the information above, <br> the majority of people in the data set are Catholic (85.6%).  </center></b>**
===========================================

**<b><u>Dataset 2: Image Report Data</u></b>**
===========================================

**<center><b>To analyze the data in question we must import the pandas library. <br> We already did this for the dataset above thus we won't need to do so, since we're in the same notebook.<center><b>**

**<center><b>Next, we will import the data from a csv named reportsubmission.csv, transform it into a dataframe, and display the items in the csv file.<center><b>**

In [7]:
reportsubmission = pd.read_csv('reportsubmission.csv')
reportsubmission = pd.DataFrame(reportsubmission)
reportsubmission

AttachmentID         FileName   FileType   MimeType  ReportID  \
0             1  BrokenGlass.jpg  Image/jpg  Image/jpg         1   
1             2    Footprint.jpg  Image/jpg  Image/jpg         2   

       Location Latitude/Longitude         Date  
0  Anytown, USA    29.531, -82.184  Mar-07-2021  
1      Colorado     29.125,-81.682       3/2/21

**<center><b>This data in question has only two rows but has alot of data which needs to be rectified immediately. <br><br> The code below will do the following consecutively:</center></b>**
<ol>
<li>Replace the items in the File Type Column with Image so it can correspond with the correct description
<br>
<br>
<li>Replace the items in the MimeType Column with Jpg so it can have the correct format
<br>
<br>
<li>Replace the Anytown, USA data in the Location Column with Florida (we will change the Location title later!)
<br>
<br>
<li>Replace the date Mar-07-2021 in the Date Column with 3/7/21 so it can have the same format as the second row
</ol> 

In [8]:
reportsubmission['FileType'].replace(['Image/jpg'],'Image', inplace=True)
reportsubmission['MimeType'].replace(['Image/jpg'],'.Jpg', inplace=True)
reportsubmission['Location'].replace(['Anytown, USA'],'Florida', inplace=True)
reportsubmission['Date'].replace(['Mar-07-2021'],'3/7/21', inplace=True)

**<center><b>Let's generste the new dataframe with the new data:<center><b>**

In [9]:
reportsubmission

AttachmentID         FileName FileType MimeType  ReportID  Location  \
0             1  BrokenGlass.jpg    Image     .Jpg         1   Florida   
1             2    Footprint.jpg    Image     .Jpg         2  Colorado   

  Latitude/Longitude    Date  
0    29.531, -82.184  3/7/21  
1     29.125,-81.682  3/2/21

**<center><b>Now let's separate the longitude and latitude field which is combined.<br> First let's insert a longtitude field after the Latitude/Longitude field with the longitude values:<center><b>**

In [10]:
reportsubmission.insert(7,'Longitude', ['-82.184', '-81.682'],True)
reportsubmission

AttachmentID         FileName FileType MimeType  ReportID  Location  \
0             1  BrokenGlass.jpg    Image     .Jpg         1   Florida   
1             2    Footprint.jpg    Image     .Jpg         2  Colorado   

  Latitude/Longitude Longitude    Date  
0    29.531, -82.184   -82.184  3/7/21  
1     29.125,-81.682   -81.682  3/2/21

**<center><b>Now let's rename 2 column names Location and Latitude/Longitude to reflect its true description based on the data: <center><b>**

In [11]:
reportsubmission.rename(columns = {'Location':'State','Latitude/Longitude':'Latitude'}, inplace = True)                               
reportsubmission

AttachmentID         FileName FileType MimeType  ReportID     State  \
0             1  BrokenGlass.jpg    Image     .Jpg         1   Florida   
1             2    Footprint.jpg    Image     .Jpg         2  Colorado   

          Latitude Longitude    Date  
0  29.531, -82.184   -82.184  3/7/21  
1   29.125,-81.682   -81.682  3/2/21

**<center><b>Next, we will update the latitude data: <center><b>**

In [12]:
reportsubmission['Latitude'].replace(['29.531, -82.184'],'29.531', inplace=True)
reportsubmission['Latitude'].replace(['29.125,-81.682'],'29.125', inplace=True)
reportsubmission

AttachmentID         FileName FileType MimeType  ReportID     State  \
0             1  BrokenGlass.jpg    Image     .Jpg         1   Florida   
1             2    Footprint.jpg    Image     .Jpg         2  Colorado   

  Latitude Longitude    Date  
0   29.531   -82.184  3/7/21  
1   29.125   -81.682  3/2/21

**<center><b>Based on the aformentioned information that was transformed we can analyze the following:</center></b>**
===========================================
<ol>
<li>How many files were submitted - 2
<br>
<br>
<li>Determine the Latitude/Longitude (location) of where a file was submitted
<br>
<br>
<li>Determine the name and type of file submitted and
<br>
<br>
<li>Determine the date a file was submitted
</ol> 

**<center><b>In my opinion this datset needs to be normalized!!!!</center></b>**
===========================================


**<b><u>Dataset 3: Disease Cases:</u></b>**
===========================================

**<center><b>To analyze the data in question we must import the pandas library. <br> We already did this for the dataset above thus we won't need to do so, since we're in the same notebook.<center><b>**

**<center><b>Next, we will import the data from a csv named diseasecases.csv, transform it into a dataframe, and display the items in the csv file.<center><b>**

In [13]:
diseasecases = pd.read_csv('diseasecases.csv')
diseasecases = pd.DataFrame(diseasecases)
diseasecases

Country  Year                 Rate
0  Afghanistan  1999       745 / 19987071
1  Afghanistan  2000      2666 / 20595360
2       Brazil  1999    37737 / 172006362
3       Brazil  2000    80488 / 174504898
4        China  1999  212258 / 1272915272
5        China  2000  213766 / 1280428583

**<center><b>Next, we will separate the cases and population into separate columns <br> by separating the information with the delimiter "/" with the following code.<center><b>**

In [14]:
disease_cases_split = diseasecases["Rate"].str.split("/", n = 1, expand = True)
diseasecases["Cases"] = disease_cases_split[0]
diseasecases["Population"] = disease_cases_split[1]
diseasecases.drop(columns =["Rate"], inplace = True)
diseasecases

Country  Year    Cases   Population
0  Afghanistan  1999     745      19987071
1  Afghanistan  2000    2666      20595360
2       Brazil  1999   37737     172006362
3       Brazil  2000   80488     174504898
4        China  1999  212258    1272915272
5        China  2000  213766    1280428583

**<center><b>Based on the aformentioned information that was transformed:</center></b>**
===========================================
<ol>
<li>The population has increased from 1999 to 2000 in all three countries
<br>
<br>
<li>China has the largest population and the most cases of diseases in 1999 and 2000
<br>
<br>
<li>There was a drastic increase in disease cases in Afganistan and Brazil from 1999 to 2000
<br>
<br>
<li>Afghanistan has the least amount of cases in both 1999 and 2000 based on the dataset
</ol> 

**<center><b>In my opinion this datset needs to be normalized as well!!!!</center></b>**
===========================================